src/main/java/cn/nukkit/utils/BinaryStream.java

```diff
@@ -528,11 +551,43 @@ public void putSlot(int protocolId, Item item) {
         }
 
         // Multiversion: Replace unsupported items
-        // TODO: Send the original item data in nbt and read it from there in getSlot, replace netherite items with diamond items for < 1.16
+        boolean needOriginalID = false;
         if (protocolId < ProtocolInfo.v1_14_0 && (networkId == Item.HONEYCOMB || (networkId == Item.SUSPICIOUS_STEW && protocolId < ProtocolInfo.v1_13_0))) {
             networkId = Item.INFO_UPDATE;
         } else if (protocolId < ProtocolInfo.v1_16_0 && networkId >= Item.LODESTONECOMPASS) {
-            networkId = Item.INFO_UPDATE;
+            needOriginalID = true;
+            switch (networkId) {
+                case Item.NETHERITE_SWORD:
+                    networkId = Item.DIAMOND_SWORD;
+                    break;
+                case Item.NETHERITE_SHOVEL:
+                    networkId = Item.DIAMOND_SHOVEL;
+                    break;
+                case Item.NETHERITE_PICKAXE:
+                    networkId = Item.DIAMOND_PICKAXE;
+                    break;
+                case Item.NETHERITE_AXE:
+                    networkId = Item.DIAMOND_AXE;
+                    break;
+                case Item.NETHERITE_HOE:
+                    networkId = Item.DIAMOND_HOE;
+                    break;
+                case Item.NETHERITE_HELMET:
+                    networkId = Item.DIAMOND_HELMET;
+                    break;
+                case Item.NETHERITE_CHESTPLATE:
+                    networkId = Item.DIAMOND_CHESTPLATE;
+                    break;
+                case Item.NETHERITE_LEGGINGS:
+                    networkId = Item.DIAMOND_LEGGINGS;
+                    break;
+                case Item.NETHERITE_BOOTS:
+                    networkId = Item.DIAMOND_BOOTS;
+                    break;
+                default:
```

Comment:
We could also have
`WARPED_FUNGUS_ON_A_STICK`  => `CARROT_ON_A_STICK`
and 
`RECORD_PIGSTEP` => `RECORD_13` (or any other record) here